rosbag2_performance/rosbag2_benchmarking/rosbag2_benchmarking/system_monitor.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2020, Robotec.ai sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""System monitor for monitoring disk, memory and cpu utilization."""
+
+import pathlib
+import re
+import signal
+import subprocess
+import threading
+
+import psutil
+
+import rclpy
+from rclpy.node import Node
+
+
+class SystemMonitor(Node):
+    """ROS2 node for monitoring disk, memory and cpu utilization."""
+
+    dstat_proc = None
+    iotop_ready = False
+
+    def __init__(self):
+        """Initialize monitor."""
+        super().__init__('system_monitor')
+        self.logger = rclpy.logging.get_logger('SYSTEM_MONITOR')
+        self.declare_parameter('benchmark_path')
+        self.declare_parameter('frequency', 10)
+
+        self.__benchmark_path = \
+            self.get_parameter('benchmark_path') \
+                .get_parameter_value().string_value
+        self.__frequency = \
+            self.get_parameter('frequency').get_parameter_value().integer_value
+
+        if self.__frequency == 0:
+            self.logger.warn(
+                'Monitor frequency is set to 0. Invalid value. \
+                Rolling back to 10Hz.'
+            )
+            self.__frequency = 10
+
+        self.create_timer(1/self.__frequency, self.gather_cpu_and_mem_data)
+        self.iotop_thread = threading.Thread(None, self.__launch_iotop)
+        self.iotop_thread.start()
+
+        bench_path = pathlib.Path(self.__benchmark_path)
+        self.__cpu_file = open(bench_path.joinpath('system_cpu.csv'), 'w')
+        self.__mem_file = open(bench_path.joinpath('system_mem.csv'), 'w')
+        self.__disk_file = open(bench_path.joinpath('system_disk.csv'), 'w')
+
+    def __launch_iotop(self):
+        self.dstat_proc = subprocess.Popen(
+            [
+                'pkexec',
+                'iotop',
+                '-b',
+                '-o',
+                '-d',
+                str(1/self.__frequency),
+                '--kilobytes'
+            ],
```

Comment:
so this wouldn't run on windows. Not that it's super important, but I feel we should have a sanity check here that the exec is actually available.

Reply:
Good point. I will add OS check here.